wagtailmedia/models.py

```diff
@@ -130,6 +150,31 @@ def get_media_model():
     return media_model
 
 
+# Receive the post_save signal and sniff mediainfo data if possible.
+@receiver(post_save, sender=Media)
+def media_sniff(sender, instance, created, update_fields, **kwargs):
+    if hasattr(settings, 'WAGTAILMEDIA_FFPROBE_CMD'):
+        if created or (update_fields and 'file' in update_fields):
+            data = sniff_media_data(instance.file.path)
+            if data:
+                duration = int(float(data['format']['duration']))
+                Media.objects.filter(pk=instance.pk).update(duration=duration, mediainfo=data)
+                if instance.type == 'video':
+                    video_stream = get_video_stream_data(data)
+                    if video_stream:
+                        height = int(float(video_stream['height']))
+                        width = int(float(video_stream['width']))
+                        Media.objects.filter(pk=instance.pk).update(height=height, width=width)
+
+                    # Try to scrape a thumbnail from video
+                    if hasattr(settings, 'WAGTAILMEDIA_FFMPEG_CMD')\
+                            and not instance.thumbnail:
```

Comment:
NB this leaves no way to leave a thumbnail blank. I'm happy with this, but just pointing it out for discussion if anyone has an opinion.

Reply:
I agree with you here. I think I'll add a "No thumbnail" checkbox to the Media admin UI.